tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -881,6 +886,172 @@ def _wait_for_traffic(self, producer, acked, timeout=1800):
                 f"Low throughput while preparing for the test: {_throughput} MB/s"
             ) from None
 
+    # returns a tuple of (replicas, readyReplicas)
+    def get_cluster_replicas(self, cluster_name: str) -> tuple[int, int]:
+        if self.operator_version == 2:
+            return self._get_cluster_replicas_v2(
+            ), self._get_cluster_ready_replicas_v2()
+        return self._get_cluster_replicas(
+            cluster_name), self._get_cluster_ready_replicas(cluster_name)
+
+    # sets new target replica count for the cluster
+    def set_cluster_replicas(self, cluster_name: str, new_replicas: int):
+        # for operator V2, we scale the statefulset because that seems
+        # to be the only way that works
+        if self.operator_version == 2:
+            self.redpanda.kubectl.cmd([
+                'scale', '-n=redpanda', 'statefulset', 'redpanda-broker',
+                f'--replicas={new_replicas}'
+            ])
+            return
+
+        # for operator v1, we patch the cluster CRD instead
+        patch = [{
+            'op': 'replace',
+            'path': '/spec/replicas',
+            'value': new_replicas
+        }]
+        patch_str = json.dumps(patch)
+
+        # kubectl patch -n=redpanda cluster rp-clkd0n22nfn1jf7vd9t0 --type=json -p='[{"op":"replace","path":"/spec/replicas","value":4}]'
+        self.redpanda.kubectl.cmd([
+            'patch', '-n=redpanda', 'cluster', cluster_name, '--type=json',
+            f"-p='{patch_str}'"
+        ])
+
+    def _kube_autoretry_int(self,
+                            command: list[str],
+                            max_retries: int = 10) -> int:
+        result = ''
+        retry_count = 0
+        while retry_count < max_retries:
+            result = self.redpanda.kubectl.cmd(command)
+            if type(result) is str and result != '':
+                return int(result)
+
+            retry_count += 1
+            self.logger.debug(
+                f'retrying `kubectl {command}` attempt #{retry_count}')
+            time.sleep(0.50)
+        raise TimeoutError(
```

Comment:
This loses the error cause. A better way to structure this is when the tries are exceeded you throw from within the loop, with the relevant error info (return code, stdout etc)

Reply:
I believe I implemented this mostly the way you meant, except `self.redpanda.kubectl.cmd` doesn't provide us the kubectl process return code.